### PR TITLE
Revamp survey selector screen

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Towards #504

https://user-images.githubusercontent.com/8918023/213907238-e37bd24f-1d0c-4cb8-8253-35a8b2b0c5d4.mp4

Next steps:
 * Progress dialog after selecting a survey
 * Reopen home screen on back press when opened via nav drawer
 * Add a toolbar / drawer icon with logo and app name

@gino-m PTAL?

### CODE REVIEW

**Comment by shobhitagarwal1612** [comment]
@gino-m Filed https://github.com/google/ground-android/labels/survey%20selector to fix the remaining improvements in followup PRs

**Comment by shobhitagarwal1612** [comment]
@gino-m Gentle ping!

File: ground/src/main/java/com/google/android/ground/MainViewModel.kt
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/MainViewModel.kt:150]: What does this method have to do with which user is signed in? Looks like it's forked based on whether there was a survey previously active.. so maybe this is `getDirectionsAfterSignIn`?

File: ground/src/main/java/com/google/android/ground/ui/surveyselector/SurveySelectorFragment.kt
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/ui/surveyselector/SurveySelectorFragment.kt:2]: Please use current copyright year for new files.

File: ground/src/main/java/com/google/android/ground/ui/surveyselector/SurveySelectorViewModel.kt
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/ui/surveyselector/SurveySelectorViewModel.kt:52]: Can we simplify how we model the survey activation flow? At first glance it seems like we shouldn't be modeling events like `attemptingToActivate` and `surveyActivated` as UI states. Maybe coroutines can help here?

File: ground/src/main/java/com/google/android/ground/ui/tos/TermsOfServiceViewModel.kt
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/ui/tos/TermsOfServiceViewModel.kt:32]: Please file an issue and reference if you plan to come back to this.

File: ground/src/test/java/com/google/android/ground/ui/surveyselector/SurveySelectorFragmentTest.kt
Comment (gino-m) [inline ground/src/test/java/com/google/android/ground/ui/surveyselector/SurveySelectorFragmentTest.kt:2]: 2023

**Review by gino-m — CHANGES_REQUESTED** [review]
A few nits. This rocks! 🎸

File: ground/src/main/java/com/google/android/ground/ui/surveyselector/SurveySelectorViewModel.kt
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/surveyselector/SurveySelectorViewModel.kt:52]: Thanks for flagging this. Will try to come up with a more elegant way of doing it.

Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/surveyselector/SurveySelectorViewModel.kt:52]: There seems to be some problem while using nested corountine. I converted the Rx code to synchronous by wrapping it inside the following code block. But, the code never reaches the next line. 

```
externalScope.launch {
      withContext(ioDispatcher) {
...
   }
}
```

Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/surveyselector/SurveySelectorViewModel.kt:52]: I refactored some code to prevent nesting of corountines and it did work.

Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/surveyselector/SurveySelectorViewModel.kt]: An alternate solution could be to just set the active survey id and then switch to home screen fragment. Then we won't have to implement a progress dialog for Survey selector fragment either. Thoughts? @gino-m

File: ground/src/main/java/com/google/android/ground/ui/tos/TermsOfServiceViewModel.kt
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/ui/tos/TermsOfServiceViewModel.kt:32]: I've been using the convention `TODO(#issue no)` since the issue link isn't clickable in code anyway, but I don't feel strongly about this.

File: ground/src/main/java/com/google/android/ground/repository/SurveyRepository.kt
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/repository/SurveyRepository.kt:130]: I actually have a big PR pending that updates the loading state here explicitly, making it a little easier to follow. I'll apply that change once I get to merging things properly.

**Review by gino-m — CHANGES_REQUESTED** [review]

File: ground/src/main/java/com/google/android/ground/ui/tos/TermsOfServiceViewModel.kt
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/tos/TermsOfServiceViewModel.kt]: Reverted for consistency.

File: ground/src/main/java/com/google/android/ground/ui/surveyselector/SurveySelectorViewModel.kt
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/surveyselector/SurveySelectorViewModel.kt]: Reverted changes to SurveyRepository

File: ground/src/main/java/com/google/android/ground/repository/SurveyRepository.kt
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/repository/SurveyRepository.kt]: Thanks for the heads up! I've reverted changes to the survey repository to unblock this PR and delegated the survey loading/activation flow to HomeScreenFragment.

File: ground/src/main/java/com/google/android/ground/ui/home/HomeScreenFragment.kt
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/ui/home/HomeScreenFragment.kt:79]: We should also probably move the progress dialog into the survey selector screen and replace with a non-deprecated component. Fyi I'll be removing this in my upcoming cleanup PR removing shared repository state.

Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/home/HomeScreenFragment.kt:79]: I was hoping to resolve this separately since the scope of this PR is to replace the dialog with this new fragment. I've already filed a bug to create a progress dialog for survey selector fragment. So, please feel free to clean this in your upcoming PR.

Comment (gino-m) [reply ground/src/main/java/com/google/android/ground/ui/home/HomeScreenFragment.kt:79]: Sgtm!

**Review by gino-m — APPROVED** [review]
📋 ✅ !

**Pull request merged**
